src/index.ts

```diff
@@ -426,6 +454,46 @@ export class UnleashClient extends TinyEmitter {
         await this.storage.save(storeKey, toggles);
     }
 
+    private isUpToDate() {
+        if (!this.experimental?.togglesStorageTTL || this.experimental?.togglesStorageTTL <= 0) {
+            return false;
+        }
+        const timestamp = Date.now();
+
+        return !!(
```

Comment:
J'aillais dire qu'il faudrait vérifier le hash ici mais en fait si je comprends bien on aura une valeur seulement si on le bon contexte ou si on s'est mis à jour dans cette "session" ?

Reply:
Sur une mise a jour de contexte, on va toujours setter le LastUpdateTerms si l’appel vers le unleash-proxy est OK. Actuellement nous n'avons pas de code qui appelle isUpToDate dans ce process car pas de raison de verifier si c'est à jour, on veut juste ecraser.

En plus, dans l’idée normalement si tu utilise une maj de contexte, c’est que tu as deja instancier un objet, et que tu l’utilise, donc il n’y a pas de raison d’avoir un contexte qui aurait bougé subitement. .. meme si c’était le cas, du coup on viendrait écraser le LastUpdateTerms lors des maj …
Le risque ça serait d’avoir plusieurs instances qui ont un contexte différent, et qui vivent leur vie, qui se décharge et se recharge …. dans ce cas la notre optim serait bcp moins pertinente lors de l'init car le LastUpdateTerms passerait son temps a switcher de hash …

Pour repondre à la question :) 
isUpToDate() n'est appelé que dans initialFetchToggles.
lorsque l'on est ici (initialFetchToggles), on est passé dans init() qui a initialisé : 
this.lastRefreshTimestamp = await this.getLastRefreshTimestamp();
uniquement si le contexte actuel match avec le contexte du lastUpdateTerms, sinon lastRefreshTimestamp est à 0